_plugins/jekyll-multiple-languages-plugin.rb

```diff
@@ -0,0 +1,67 @@
+module Jekyll
+
+
+  ##############################################################################
+  # class Site
+  ##############################################################################
+  class Site
+
+    attr_accessor :translation_file_content   # Hash that stores parsed translations read from YAML files.
+    alias_method :process_extended, :process
+
+    def process
+      self.translation_file_content ||= {}
+      process_extended
+    end
+  end
+
+  module LocalizeTagExtensions
+    #======================================
+    # initialize
+    #======================================
+    def initialize(tag_name, key, tokens)
+      super
+      @key = key.strip
+    end
+    #======================================
+    # render
+    #======================================
+    def render(context)
+      if      "#{context[@key]}" != "" # Check for page variable
+        key = "#{context[@key]}"
+      else
+        key =            @key
+      end
+      
+      site = context.registers[:site] # Jekyll site object
+      
+      lang = site.config['lang']
+      
+      # unless site.parsed_translations.has_key?(lang)
+      #   puts              "Loading translation from file #{site.source}/_i18n/#{lang}.yml"
+      #   site.parsed_translations[lang] = YAML.load_file("#{site.source}/_i18n/#{lang}.yml")+      
+
+      unless site.parsed_translations.has_key?(lang)
+        puts "Loading translation from file #{site.source}/_i18n/#{lang}.yml"
+        site.translation_file_content = YAML.load_file("#{site.source}/_i18n/#{lang}.yml")
+        site.parsed_translations[lang] = site.translation_file_content[lang]
+      end
+
+      
+      translation = site.parsed_translations[lang].access(key) if key.is_a?(String)
```

Comment:
I really prefer to avoid `is_a?` if possible as in Ruby it's more of a convention to use Duck typing and checking specific types can be considered an anti-pattern.

Is it possible that `key` can be anything other than a String here?